### PR TITLE
Fix low level interpreter tests johanandren

### PR DESCRIPTION
**Pull request by johanandren**
Refs #22423 

Was thinking maybe we could merge this and work on the few disabled tests in it separately to get some test coverage back in place.

### CODE REVIEW

**Comment by johanandren** [comment]
Btw, github PR-diff is completely hiding the big actual change for some reason. Is in `GraphInterpreterSpecKit`

**Comment by johanandren** [comment]
Failures are in akka-bench-jmh and akka-docs (not related to this pr)

**Comment by johanandren** [comment]
Ok, after team discussion there is no need to urgently merge this as the failures in these tests are only about the GraphAssembly not being in there anymore and not because of changes in the interpreter/materializer.

**Comment by johanandren** [comment]
Should be done for review now.

**Review by patriknw — APPROVED** [review]
LGTM, I guess

**Comment by johanandren** [comment]
I'll push a commit making the jmh and docs compile as well.

**Comment by 2m** [comment]
Great. Maybe after that I will be able to run benchmarks from the target branch without any mods.

File: akka-stream-tests/src/test/scala/akka/stream/impl/GraphStageLogicSpec.scala [hunk 1]
```diff
@@ -55,28 +54,35 @@ class GraphStageLogicSpec extends StreamSpec with GraphInterpreterSpecKit {
   object passThrough extends GraphStage[FlowShape[Int, Int]] {
     val in = Inlet[Int]("in")
     val out = Outlet[Int]("out")
+    in.id = 0
+    out.id = 1
```
Comment (drewhk) [inline akka-stream-tests/src/test/scala/akka/stream/impl/GraphStageLogicSpec.scala]: it should be 0 shouldn't it? There is only one output.

Comment (drewhk) [inline akka-stream-tests/src/test/scala/akka/stream/impl/GraphStageLogicSpec.scala]: Yes it should be 0. Look at `TraversalBuilder.initShape()`

Comment (johanandren) [inline akka-stream-tests/src/test/scala/akka/stream/impl/GraphStageLogicSpec.scala]: Yeap. It is re-enumerated in the create logics though, this was left over from the early trial and error.

**Comment by drewhk** [comment]
@johanandren don't waste time on NewLayoutBenchmark, just remove it. The only reason it exists because I wanted to measure things before I had wired up everything to the DSL.

**Comment by drewhk** [comment]
I.e. the DSL was simply not available for that purpose back then.

**Comment by johanandren** [comment]
I think we can merge this now.

**Comment by drewhk** [comment]
Worky worky now?

**Comment by johanandren** [comment]
Yeah, the failures are unrelated (mima and other test fails etc)

**Comment by johanandren** [comment]
Mergy merge.

**Comment by patriknw** [comment]
Great work, @johanandren !

File: akka-remote/src/main/scala/akka/remote/serialization/PrimitiveSerializers.scala [hunk 2]
```diff
@@ -84,7 +84,10 @@ class StringSerializer(val system: ExtendedActorSystem) extends BaseSerializer w
     new String(bytes, "UTF-8")
   }
 
-  override def toBinary(o: AnyRef): Array[Byte] = o.asInstanceOf[String].getBytes("UTF-8")
+  override def toBinary(o: AnyRef): Array[Byte] = {
+    println("stringserializer")
+    o.asInstanceOf[String].getBytes("UTF-8")
+  }
```
Comment (2m) [inline akka-remote/src/main/scala/akka/remote/serialization/PrimitiveSerializers.scala]: Some remains of the serialization work?

File: akka-remote/src/test/scala/akka/remote/serialization/DaemonMsgCreateSerializerSpec.scala [hunk 3]
```diff
@@ -91,6 +91,9 @@ class DaemonMsgCreateSerializerSpec extends AkkaSpec {
         DaemonMsgCreate(Props(classOf[MyActorWithParam], "a string"), Deploy.local, "/user/test", system.actorFor("/user")))
       println(String.valueOf(encodeHex(bytes)))
       */
+      import org.apache.commons.codec.binary.Hex.encodeHex
+
+      println(String.valueOf(encodeHex(SerializationExtension(system).serialize("a string").get)))
```
Comment (2m) [inline akka-remote/src/test/scala/akka/remote/serialization/DaemonMsgCreateSerializerSpec.scala]: Some remains of the serialization work?

File: akka-remote/src/main/scala/akka/remote/serialization/PrimitiveSerializers.scala (hunk 2 above)
Comment (johanandren) [inline akka-remote/src/main/scala/akka/remote/serialization/PrimitiveSerializers.scala]: Yeah, ups. I'll make a new pr fixing. Good catch.

**Pull request merged**